Common/Algorithm/Framework/Alphas/OrderBasedInsightGenerator.cs

```diff
@@ -63,19 +63,20 @@ public class OrderBasedInsightGenerator
             else
             {
                 // we are reducing the position, so set the confidence based on the original position
-                confidence = (double)(securityHolding.AbsoluteQuantity - orderEvent.AbsoluteFillQuantity)
-                    / (double) securityHolding.AbsoluteQuantity;
+                confidence = 1 - (double) (orderEvent.AbsoluteFillQuantity / Math.Abs(preFillQuantity));
 
                 if (existingInsight != null)
                 {
                     // we have to adjust new confidence based on previous
-                    confidence = confidence * existingInsight.Confidence;
+                    confidence *= existingInsight.Confidence;
                 }
             }
 
-            var insightDirection = desiredFinalQuantity > 0
-                ? InsightDirection.Up : desiredFinalQuantity == 0
-                    ? InsightDirection.Flat : InsightDirection.Down;
+            var insightDirection = securityHolding.IsLong
```

Comment:
no changed required -- how to be an [`InsightDirection`](https://github.com/QuantConnect/Lean/blob/master/Common/Algorithm/Framework/Alphas/InsightDirection.cs#L35) setting boss -- `InsightDirection` is defined using -1, 0, and 1. This was done to allow us to simplify some directional maths, _but also_ to simplify setting the insight's direction using signed maths:

``` csharp
// https://docs.microsoft.com/en-us/dotnet/api/system.math.sign?view=netframework-4.8
var insightDirection = (InsightDirection)Math.Sign(securityHolding.Quantity)
```